src/main/java/duke/command/CreateCommand.java

```diff
@@ -0,0 +1,112 @@
+        if (this.taskType == TaskType.TODO) {
+            this.parseCreateTodo(linePrinter, taskList);
+        } else if (this.taskType == TaskType.DEADLINE) {
+            this.parseCreateDeadline(linePrinter, taskList);
+        } else if (this.taskType == TaskType.EVENT) {
+            this.parseCreateEvent(linePrinter, taskList);
+        }
+
+        return true;
+    }
+
+    /**
+     * Parses and creates a Todo object.
+     * @param linePrinter Object that the command handler outputs responses to.
+     * @param taskList Current task list that the command handler should add a new task to.
+     * @throws DukeIllegalArgumentException If the argument string is empty.
+     */
+    private void parseCreateTodo(Printable linePrinter, TaskList taskList)
+            throws DukeIllegalArgumentException {
+        // Syntax Check
+        if (args.equals("")) {
+            throw new DukeIllegalArgumentException("Task name cannot be empty");
+        }
+
+        final Task task = taskList.addTask(new Todo(this.args));
+        this.printResponse(linePrinter, task, taskList, "Todo");
+    }
+
+    /**
+     * Parses and creates a Deadline object.
+     * @param linePrinter Object that the command handler outputs responses to.
+     * @param taskList Current task list that the command handler should add a new task to.
+     * @throws DukeIllegalArgumentException If the argument string does not follow the expected format.
+     */
+    private void parseCreateDeadline(Printable linePrinter, TaskList taskList)
+            throws DukeIllegalArgumentException {
+        // Syntax Check
+        final String[] argParts = this.args.split(" /by ");
+        if (argParts.length < 2) {
+            throw new DukeIllegalArgumentException("Not in the format <Task name> /by <Date>");
+        }
+
+        final String taskDescription = argParts[0];
+        if (taskDescription.equals("")) {
+            throw new DukeIllegalArgumentException("Task name cannot be empty");
+        }
+        final LocalDateTime taskBy = parseDateTime(argParts[1]);
+        final Task task = taskList.addTask(new Deadline(taskDescription, taskBy));
+        this.printResponse(linePrinter, task, taskList, "Deadline");
+    }
+
+    /**
+     * Parses and creates a Event object.
+     * @param linePrinter Object that the command handler outputs responses to.
+     * @param taskList Current task list that the command handler should add a new task to.
+     * @throws DukeIllegalArgumentException If the argument string does not follow the expected format.
+     */
+    private void parseCreateEvent(Printable linePrinter, TaskList taskList)
+            throws DukeIllegalArgumentException {
+        // Syntax Check
+        final String[] argParts = this.args.split(" /at ");
+        if (argParts.length < 2) {
+            throw new DukeIllegalArgumentException("Not in the format <Task name> /at <Date>");
```

Comment:
You have really excellent detailed error messages provided to the user! I'll learn from this thanks! 😃